libs/dal/src/shared/interceptors/shared-cache.interceptor.ts

```diff
@@ -0,0 +1,38 @@
+function validateCredentials(keyToValidate: string) {
+  const splitKeys = keyToValidate?.split(':');
+
+  return splitKeys.length === 3;
+}
+
+const STORE_CONNECTED = 'ready';
+
+export function isStoreConnected(status: string | undefined) {
+  return status === STORE_CONNECTED;
+}
+
+/**
+ * Will append the identifier and environmentId, if failed to append one of them will return empty string
+ * @param key
+ * @param keyConfig
+ */
+export function appendCredentials(key: string, keyConfig: Record<string, undefined>) {
```

Comment:
Would be great to unit test those functions, should be quite straight forward and benfitial i believe.